src/components/ListItem.jsx

```diff
@@ -1,5 +1,45 @@
 import './ListItem.css';
+import { updateItem } from '../api/firebase';
+import { useEffect, useState } from 'react';
+import { getFutureDate } from '../utils';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ name, listToken, itemId, data }) {
+	const [checkedState, setCheckedState] = useState(data.checked);
+
+	const checkTimePast = () => {
+		if (data.dateLastPurchased !== null) {
+			const yesterday = getFutureDate(-1);
```

Comment:
```suggestion
			const yesterday = getFutureDate(-1);
```
if this was showing the older code not using the getFutureDate method this demo of getFutureDate would make more sense